src/fmpy/fmi3.py

```diff
@@ -775,9 +790,16 @@ def setString(self, vr, values):
         values = (fmi3String * len(values))(*values)
         self.fmi3SetString(self.component, vr, len(vr), values, len(values))
 
-    def setBinary(self, vr, values):
+    def setBinary(self, vr: Iterable[int], values: Iterable[bytes]):
+
         vr = (fmi3ValueReference * len(vr))(*vr)
-        values_ = (fmi3Binary * len(values))(*values)
+
+        values_ = (fmi3Binary * len(values))()
+
+        for i, v in enumerate(values):
+            b = (c_uint8 * len(v)).from_buffer(bytearray(v))
+            values_[i] = b
+
         size = (c_size_t * len(vr))(*[len(v) for v in values])
```

Comment:
Should be ` size = (c_size_t * len(values))(*[len(v) for v in values])` to cope with Binary variables that are arrays